python/lsst/analysis/drp/histPlot.py

```diff
@@ -0,0 +1,387 @@
+                      self.config.rightHistColumns))
+
+        fig = plt.figure()
+        gs = gridspec.GridSpec(100, 100)
+
+        # left panel limits
+        pLowers, pUppers = [], []
+        for key in self.config.leftHistColumns.keys():
+            column = self.config.leftHistColumns[key]
+            [pLower, pUpper] = np.nanpercentile(catPlot[column].values,
+                                                [self.config.pLower, self.config.pUpper])
+            pLowers.append(pLower)
+            pUppers.append(pUpper)
+        leftHistLower, leftHistUpper = np.min(pLowers), np.max(pUppers)
+
+        # left panel plotting
+        leftHistAx = fig.add_subplot(gs[:, 0:35])
+        for count, key in enumerate(self.config.leftHistColumns.keys()):
+            column = self.config.leftHistColumns[key]
+            mask = np.isfinite(catPlot[column].values)
+            label = f"{column} ({np.sum(mask)})"
+            col = plt.cm.tab10(count)
+            leftHistAx.hist(catPlot[column][mask], bins=self.config.nBins, alpha=0.7, color=col,
+                            range=(leftHistLower, leftHistUpper), histtype="bar", lw=2, label=label)
+            datMed = np.nanmedian(catPlot[column][mask])
+            datMad = sigmaMad(catPlot[column][mask])
+            leftHistAx.text(x=0.03, y=0.985-(count*0.08), transform=leftHistAx.transAxes,
+                            ha="left", va="top", fontsize=7, c=col,
+                            s=f"med = {datMed:0.2f}\n$σ_{{MAD}}$ = {datMad:0.2f}")
+            leftHistAx.axvline(datMed, ls=':', lw=2, c=col)
+        leftHistAx.set_xlim(leftHistLower, leftHistUpper)
+        leftHistAx.set_xlabel(self.config.axisLabels["xLeft"])
+        leftHistAx.set_ylabel(self.config.axisLabels["yLeft"])
+        leftHistAx.legend(loc="lower right", bbox_to_anchor=(0.995, 0.1),
+                          bbox_transform=fig.transFigure, fontsize=8)
+        leftHistAx.axvline(0, ls='--', lw=2, c='k')
+
+        # right panel limits
+        pLowers, pUppers = [], []
+        for key in self.config.rightHistColumns.keys():
+            column = self.config.rightHistColumns[key]
+            [pLower, pUpper] = np.nanpercentile(catPlot[column].values,
+                                                [self.config.pLower, self.config.pUpper])
+            pLowers.append(pLower)
+            pUppers.append(pUpper)
+        rightHistLower, rightHistUpper = np.min(pLowers), np.max(pUppers)
+
+        # right panel plotting
+        rightHistAx = fig.add_subplot(gs[:, 43:78])
+        for count, key in enumerate(self.config.rightHistColumns.keys()):
+            column = self.config.rightHistColumns[key]
+            mask = np.isfinite(catPlot[column].values)
+            col = plt.cm.tab10(count)
+            rightHistAx.hist(catPlot[column][mask], bins=self.config.nBins, alpha=0.7, color=col,
+                             range=(rightHistLower, rightHistUpper), histtype="bar", lw=2)
```

Comment:
You don't have labels or a legend for the right-hand histogram, because it is assumed that the data will correspond to the same measurements (i.e. PSFlux and 9ApFlux here), but could the user run into problems by setting different measurements? Also, if they list dictionary items differently in `leftHistColumns` and `rightHistColumns`, I think you would end up having the colors switched.

Reply:
See main reply comment below which addresses the main issue regarding naming conventions/restrictions. With regards no label for the secondary y-axis, this is intentional due to the limited space on the horizontal axis. As both panels will always display a (number-count) histogram, the y-axes should always be some measure of counts/frequency, hence a unified y-axis naming convention here. 

It's not impossible to imagine switching one of the panels from displaying frequency/number histograms to density histograms instead, however, the code as-is currently isn't set up to construct such a plot, so this point is moot. If it's okay with you, I suggest putting off adding an additional y-axis label until such time as one is needed, preserving as much horizontal space as possible for figure plotting.